File: redsheet-portal/src/app/database-models/v-role.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vRole {
    constructor(vRole?: vRole )  {
        let properties = [ "roleId", "roleName", "status", "statusId",];
        for (let property of properties)
            if (vRole && vRole[property])
                this[property] = vRole[property];
    }
 
	roleId : number =   0 ;
 
    @required()
    @maxLength(50)
	roleName : string =   undefined;
 
    @required()
    @maxLength(100)
	status : string =   undefined;
 
    @range(1,2147483647)
	statusId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vRole.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vRoles", Schema = "dbo")]
	[Serializable]
	public partial class vRole  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int RoleId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string RoleName { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Status { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int StatusId { get; set; } 

	public vRole()
	{
    }

    }
}

